File: 0513-find-bottom-left-tree-value/0513-find-bottom-left-tree-value.java
class Solution {
    public int findBottomLeftValue(TreeNode root) {
        int result = 0;
        Queue<TreeNode> que = new ArrayDeque<>();

        que.offer(root);

        while (!que.isEmpty()) {
            int n = que.size();

            for (int i = 0; i < n; i++) {
                if (i == 0) {
                    result = que.peek().val;
                }
                if (que.peek().left != null) {
                    que.offer(que.peek().left);
                }
                if (que.peek().right != null) {
                    que.offer(que.peek().right);
                }
                que.poll();
            }
        }

        return result;
    }
}

File: 0513-find-bottom-left-tree-value/0513-find-bottom-left-tree-value.cpp
class Solution {
public:
    int findBottomLeftValue(TreeNode* root) {
        int result = 0;
        queue<TreeNode*> que;

        que.push(root);

        while (!que.empty()) {
            int n = que.size();

            for (int i = 0; i < n; i++) {
                if (i == 0) {
                    result = que.front()->val;
                }
                if (que.front()->left != NULL) {
                    que.push(que.front()->left);
                }
                if (que.front()->right != NULL) {
                    que.push(que.front()->right);
                }
                que.pop();
            }
        }

        return result;
    }
};